src/slurm_plugin/clustermgtd.py

```diff
@@ -749,142 +660,37 @@ def _update_static_nodes_in_replacement(self, slurm_nodes):
                 nodes_still_in_replacement.add(nodename)
         self._static_nodes_in_replacement = nodes_still_in_replacement
 
-    def _group_slurm_nodes(self, slurm_nodes, private_ip_to_instance_map):
+    def _group_slurm_nodes(self, slurm_nodes):
         """
         Group slurm nodes into different types.
 
         Check and return slurm nodes with unhealthy and healthy scheduler state, grouping unhealthy nodes
         by node type (static/dynamic).
         """
+        powering_down_nodes = []
         unhealthy_static_nodes = []
         unhealthy_dynamic_nodes = []
         bootstrap_failure_nodes = []
         healthy_nodes = []
         for node in slurm_nodes:
-            if not self._is_node_healthy(node, private_ip_to_instance_map):
-                if node.is_static:
+            if node.is_powering_down_node_with_nodeaddr():
+                powering_down_nodes.append(node)
+            if not node.is_healthy(self._config.terminate_drain_nodes):
```

Comment:
reminder that `is_healthy` might need to depend on both `terminate_drain_nodes` and `terminate_down_nodes`